scopes/react/react-native/webpack/webpack-transformers.ts

```diff
@@ -1,17 +1,31 @@
 import { WebpackConfigTransformer, WebpackConfigMutator, WebpackConfigTransformContext } from '@teambit/webpack';
 
+const reactNativePackagesRule = {
+  test: /\.(jsx?|tsx?)$/,
+  include: [/node_modules\/react-native-/],
+  loader: require.resolve('babel-loader'),
+  options: {
+    cacheDirectory: false,
+    presets: ['@babel/preset-env', '@babel/preset-react'],
+  },
+};
+
+
 /**
  * Transformation to apply for both preview and dev server
  * @param config
  * @param _context
  */
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 function commonTransformation(config: WebpackConfigMutator, _context: WebpackConfigTransformContext) {
-  config.addAliases({
-    react: require.resolve('react'),
-    'react-dom/server': require.resolve('react-dom/server'),
-    'react-native$': require.resolve('react-native-web'),
-  });
+  config
+    .addAliases({
+      react: require.resolve('react'),
+      'react-dom/server': require.resolve('react-dom/server'),
+      'react-native$': require.resolve('react-native-web'),
+    })
+    .addModuleRule(reactNativePackagesRule);
```

Comment:
@benjgil 
I'm not sure this will work for you as expected, as it will add it after the one of rule, which means both rules will be applied

Reply:
That's fine. 
This rule only runs on packages in node_modules, which are excluded from the oneOf babel-loader rule